### ISSUE TITLE
Bug : carriage router is breaking the tokenizer

### ISSUE THREAD

**Issue by abarre** [jakubpawlowicz/clean-css#1000]
Our solution minifies CSS from various websites and some CSS have carriage return (`\r`) without the new line character (`\n`). This character breaks the tokenizer. The only patch I found is removing all (\r) with a replace statement before applying clean-css.

### Environment

* clean-css version - `npm ls clean-css`: clean-css@4.1.9
* node.js version - `node -v`: v8.9.4
* operating system: Mac OS

### Configuration options

```js
var CleanCSS = require('clean-css');
new CleanCSS({
level: { 1: {all:false}}
})
```

### Input CSS
```css
\r@media only screen and (max-width: 1250px) {#aspot-standard {width: 100%;}}
```
**Please note the '\r' before the '@'.**

### Actual output CSS

```css
@media only screen and (max-width: 1250px){}
```

### Expected output CSS

```css
@media only screen and (max-width: 1250px){#aspot-standard{width:100%}}
```

**Comment by abarre** [comment]
congrats for the 1000th issues :)

**Comment by jakubpawlowicz** [comment]
I wonder what would happen to source maps when getting rid of `\r`, but that's not a valid line break, or is it? If not then we can bring it in, but more likely by detecting those faulty carriage returns in the tokenizer itself.

p.s. I would rather have less issues but hey let's 🎉

**Comment by abarre** [comment]
`\r` alone is a valid new line as per the [CSS spec](https://www.w3.org/TR/css-syntax-3/#newline-diagram). However, I agree with you that this is really unexpected case.

**Comment by jakubpawlowicz** [comment]
So we should handle it.

**Comment by jakubpawlowicz** [comment]
To be released with 4.2.0 tomorrow.
